### PR TITLE
fix: updating enterprise catalog api docs relating to courses, course runs and programs

### PR DESCRIPTION
**Pull request by alex-sheehan-edx**
https://2u-internal.atlassian.net/browse/ENT-8159

Add a description of your changes with links to any relevant material.

### Date Needed (optional)

If the release date of a feature is known or estimated, provide it to give reviewers guidance on turnaround time.

### Reviewers

Possible roles follow. The PR submitter checks the boxes after each reviewer finishes and gives :+1:.

- [ ] Subject matter expert:
- [ ] Subject matter expert:
- [ ] Product review:
- [ ] Partner support:
- [ ] PM review:

FYI: Tag anyone else who might be interested in this PR here.

### Testing

- [ ] Ran ./run_tests.sh without warnings or errors

### HTML Version (optional)

- [ ] Build an RTD draft for your branch and add a link here

### Sandbox (optional)

- [ ] Point to or build a sandbox for the software change and add a link here

### Post-review

- [ ] Add a comment with the description of this change or link this PR to the next release notes task.
- [ ] Squash commits

### CODE REVIEW

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 1]
```diff
@@ -581,79 +581,196 @@ Fields in a course Content Item
...
+     - UUID associated with the course type object.
+   * - ``url_slug``
+     - string
+     - The course identifying slug used in related url paths
+   * - ``course_type``
+     - string
+     - The course type. Example: ``verified-audit``.
+   * - ``enterprise_subscription_inclusion``
+     - boolean
+     - Whether the author of the course indicates the course is intended to be
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: This field signifies if this course is in the enterprise subscription catalog.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 2]
```diff
@@ -581,79 +581,196 @@ Fields in a course Content Item
...
+       included in enterprise subscriptions
+   * - ``excluded_from_seo``
+     - boolean
+     - Whether or not the course metadata is excluded from SEO
+   * - ``excluded_from_search`` 
+     - boolean
+     - Whether or not the course is searchable by the various enterprise
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: If checked, this item will not show up in search results.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 3]
```diff
@@ -581,79 +581,196 @@ Fields in a course Content Item
...
+   * - ``course_type``
+     - string
+     - The course type. Example: ``verified-audit``.
+   * - ``enterprise_subscription_inclusion``
+     - boolean
+     - Whether the author of the course indicates the course is intended to be
+       included in enterprise subscriptions
+   * - ``excluded_from_seo``
+     - boolean
+     - Whether or not the course metadata is excluded from SEO
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: If checked, the About Page will be excluded from indexing.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 4]
```diff
@@ -668,116 +785,171 @@ Fields in a course_run Content Item
...
+     - string
+     - The UUID of the associated course.
+   * - ``estimated_hours``
+     - float
+     - Estimated number of hours it takes to complete the course.
+   * - ``content_language_search_facet_name``
+     - string
+     - The language associated with the course run that is indexed for search
+       throughout the enterprise product.
+   * - ``enterprise_subscription_inclusion``
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst:942]: I think this is duplicated above (line 622)

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 5]
```diff
@@ -668,116 +785,171 @@ Fields in a course_run Content Item
...
+     - boolean
+     - Whether the author of the course indicates the course is intended to be
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: Indicates whether the course run is included in the Enterprise Subscription catalog.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 6]
```diff
@@ -668,116 +785,171 @@ Fields in a course_run Content Item
...
-   * - ``weeks_to_complete``
+     - Additional information on ofac restrictions relating to the course run.
+   * - ``enrollment_count``
      - integer
-     - The number of weeks required to complete the content item.
-
+     - Total number of learners who have enrolled in this course run.
+   * - ``recent_enrollment_count``
+     - integer
+     - Total number of learners who have enrolled in this course in the last 6
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: *course run

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 7]
```diff
@@ -668,116 +785,171 @@ Fields in a course_run Content Item
...
+     - string
+     - The language associated with the course run that is indexed for search
+       throughout the enterprise product.
+   * - ``enterprise_subscription_inclusion``
+     - boolean
+     - Whether the author of the course indicates the course is intended to be
+       included in enterprise subscriptions
+   * - ``programs``
+     - array (obj)
+     - Any programs that the course is part of.
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: An array of programs that the course is associated with.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 8]
```diff
@@ -668,116 +785,171 @@ Fields in a course_run Content Item
...
+   * - ``course_uuid``
+     - string
+     - The UUID of the associated course.
+   * - ``estimated_hours``
+     - float
+     - Estimated number of hours it takes to complete the course.
+   * - ``content_language_search_facet_name``
+     - string
+     - The language associated with the course run that is indexed for search
+       throughout the enterprise product.
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: throughout the platform (aka I don't think this field is Enterprise specific).

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 9]
```diff
@@ -668,116 +785,171 @@ Fields in a course_run Content Item
...
+       months.
+   * - ``expected_program_type``
+     - obj
+     - Designated program type record for the course run.
+   * - ``expected_program_name``
+     - string
+     - Designated expected program name for the course run.
+   * - ``course_uuid``
+     - string
+     - The UUID of the associated course.
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: The UUID of the course object associated with this course run object. 

feel free to word smith this

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 10]
```diff
@@ -656,79 +656,195 @@ Fields in a course Content Item
...
+     - Additional information relating to the course in HTML form. This 
+       information is only provided by administrators, not course authors,
+       and as such may hold special HTML that is normally not allowed.  
+   * - ``additional_metadata``
+     - obj
+     - Additional course metadata associated with 2U courses external to the
+       edX platform. 
+   * - ``faq``
+     - string
+     - HTML representation of the course FAQ section``
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: HTML representation of the course FAQ section.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 11]
```diff
@@ -656,79 +656,195 @@ Fields in a course Content Item
...
+   * - ``learner_testimonials``
+     - string
+     - HTML representation of hte course learner testimonials section
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: HTML representation of hte course learner testimonials section.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 12]
```diff
@@ -656,79 +656,195 @@ Fields in a course Content Item
...
+   * - ``enrollment_count``
+     - integer
+     - Total number of learners who have enrolled in this course.
+   * - ``recent_enrollment_count``
+     - integer
+     - Total number of learners who have enrolled in this course in the last 6
+       months.
+   * - ``topics``
+     - array (obj)
+     - Topics associated with the course
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: Topics associated with the course.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 13]
```diff
@@ -656,79 +656,195 @@ Fields in a course Content Item
...
+   * - ``key_for_reruns``
+     - string
+     - Course author provided key that is used for all reruns of the course.
+   * - ``url_slug_history``
+     - array (string)
+     - List of course slugs used for the course throughout its lifespan.
+   * - ``url_redirects``
+     - array (string)
+     - List of course url redirects
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: List of course url redirects.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 14]
```diff
@@ -656,79 +656,195 @@ Fields in a course Content Item
...
+     - Geographic location for the course, if one exists.
+   * - ``location_restriction``
+     - obj
+     - Course location restriction record.
+   * - ``in_year_value``
+     - obj
+     - Record related to projected value for a course.
+   * - ``product_source``
+     - obj
+     - Record related to course origin
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: Record related to course origin.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 15]
```diff
@@ -656,79 +656,195 @@ Fields in a course Content Item
...
+   * - ``product_source``
+     - obj
+     - Record related to course origin 
+   * - ``data_modified_timestamp``
+     - datetime
+     - The timestamp of the last time the course data was modified.
+   * - ``watchers``
+     - array (string)
+     - The list of email addresses that will be notified if any of the course
+       runs is published or scheduled.
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: The list of email addresses that will be notified if any of the course
       runs are published or scheduled.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 16]
```diff
@@ -656,79 +656,195 @@ Fields in a course Content Item
...
+   * - ``programs``
+     - array (obj)
+     - Any programs that the course is part of.
+   * - ``course_run_keys``
+     - array (string)
+     - The list of associated course run keys.
+   * - ``editable``
+     - boolean
+     - Whether the course is editable
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: Whether the course is editable.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 17]
```diff
@@ -743,116 +859,171 @@ Fields in a course_run Content Item
...
+     - An identifier for the course run. For example, ``course-v1:RITx+PM9003x``. 
+   * - ``uuid``
+     - string
+     - The unique identifier for the course run. Example: ``0dbd8181-8866-47fc...`` 
+   * - ``title``
+     - string
+     - The title of the content item. For example, "Introduction to Plasma Physics".
+   * - ``external_key``
+     - string
+     - Content item unique identifying key external to the edX platform
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: Content item unique identifying key external to the edX platform.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 18]
```diff
@@ -743,116 +859,171 @@ Fields in a course_run Content Item
...
+     - The About page image for the course. 
+   * - ``short_description``
+     - string
+     - The short description of the content item and its content. 
+   * - ``marketing_url``
+     - string
+     - The URL for the course About page.
+   * - ``seats``
+     - array (obj)
+     - List of seat records available for enrollment for the course run
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: List of seat records available for enrollment for the course run.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 19]
```diff
@@ -743,116 +859,171 @@ Fields in a course_run Content Item
...
+   * - ``start``
+     - datetime
+     - The start time for the content item.
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: The start date of the course run.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 20]
```diff
@@ -743,116 +859,171 @@ Fields in a course_run Content Item
...
+       Learners cannot enroll after this date and time.
+   * - ``weeks_to_complete``
+     - integer
+     - The number of weeks required to complete the content item.
+   * - ``pacing_type``
+     - enum string
+     - The pacing of the course. May be ``self-paced`` or ``instructor-paced``.
+   * - ``type``
+     - enum string
+     - One of ``verified``, ``professional``
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: This one seems unfinished

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 21]
```diff
@@ -743,116 +859,171 @@ Fields in a course_run Content Item
...
-     - array
-     - The unique identifiers of the subject categories of the content item.
-   * - ``title``
+     - Whether the course run has been hidden by the authors or administrators.
+   * - ``reporting_type``
+     - string
+     - Reporting type designated for the course. Example: ``mooc``.
+   * - ``eligible_for_financial_aid``
+     - bool
+     - Whether the course run is eligible to learner for financial aid.
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: Whether the course run is eligible for financial aid to a learner.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 22]
```diff
@@ -743,116 +859,171 @@ Fields in a course_run Content Item
...
+   * - ``first_enrollable_paid_seat_price``
+     - integer
+     - Available seat price for enrollment in the course.
+   * - ``has_ofac_restrictions``
+     - boolean
+     - Whether the course run has ofac restrictions, ie geographical 
+       restrictions as to where the course run can be sold.
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst:989]: Whether the course run has OFAC restrictions, i.e. geographical restrictions as to where the course run can be sold.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 23]
```diff
@@ -743,116 +859,171 @@ Fields in a course_run Content Item
...
-     - The title of the content item. For example, "Introduction to Plasma
-       Physics".
-   * - ``transcript_languages``
-     - array
-     - The languages for which video transcripts are available.
-   * - ``type``
-     - enum string
-     - One of ``verified``, ``professional``
-   * - ``weeks_to_complete``
+     - Additional information on ofac restrictions relating to the course run.
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst:992]: Additional information on OFAC restrictions relating to the course run.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 24]
```diff
@@ -743,116 +859,171 @@ Fields in a course_run Content Item
...
+     - float
+     - Estimated number of hours it takes to complete the course.
+   * - ``content_language_search_facet_name``
+     - string
+     - The language associated with the course run that is indexed for search
+       throughout the platform.
+   * - ``enterprise_subscription_inclusion``
+     - boolean
+     - Signifies whether the course run is included in the Enterprise Subscription catalog.
+       included in enterprise subscriptions
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: cut what comes after the full sentence

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 25]
```diff
@@ -743,116 +859,171 @@ Fields in a course_run Content Item
...
+   * - ``programs``
+     - array (obj)
+     - An array of programs that the course is associated with.
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: An array of programs that the course run is associated with.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 26]
```diff
@@ -868,157 +1039,739 @@ A program is a collection of related courses.
...
-     - Whether the content item has been published.
-   * - ``staff_uuids``
-     - array
-     - The unique identifiers of the staff for the program.
-   * - ``subject_uuids``
-     - array
-     - The unique identifiers of the subject categories of the program.
-   * - ``subtitle``
+     - Whether the program and program content has ofac restrictions, ie 
+       geographical restrictions as to where the course run can be sold.
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst:1142]: Whether the program and program content has OFAC restrictions, i.e. 
       geographical restrictions as to where the course run can be sold.

File: en_us/enterprise_api/source/api_reference/reference.rst [hunk 27]
```diff
@@ -868,157 +1039,739 @@ A program is a collection of related courses.
...
+   * - ``video``
+     - obj
+     - The program About video record.
+   * - ``expected_learning_items``
+     - array (obj)
+     - Elements of the course learning items records.
+   * - ``faq``
+     - array (string)
+     - List of HTML representations of the course FAQ sections under the
+       program
```
Comment (marianahernandez96) [inline en_us/enterprise_api/source/api_reference/reference.rst]: List of HTML representations of the course FAQ sections under the program.

File: en_us/enterprise_api/source/api_reference/reference.rst (hunk 20 above)
Comment (alex-sheehan-edx) [reply en_us/enterprise_api/source/api_reference/reference.rst]: oops lmao

**Review by johnnagro — APPROVED** [review]

**Review by sameenfatima78 — APPROVED** [review]

**Pull request merged**